{
   "active_lock_reason" : null,
   "assignee" : null,
   "assignees" : [],
   "author_association" : "MEMBER",
   "body" : "\r\n* Requested in [#28391 comment](https://github.com/bitcoin/bitcoin/pull/28391#discussion_r1382997719)\r\n\r\n* The error message is static when a new transaction is created and package limit is reached.\r\n`Transaction has too long of a mempool chain`\r\nWhile the [`CTxMempool::CheckPackageLimits`](https://github.com/bitcoin/bitcoin/blob/5800c558eb5efb4839ed00d6967e43306d68e1c3/src/txmempool.cpp#L199) provide explicit information about the error message.\r\n* This PR updates [`CTxMempool::CheckPackageLimits`](https://github.com/bitcoin/bitcoin/blob/5800c558eb5efb4839ed00d6967e43306d68e1c3/src/txmempool.cpp#L199) return type to `util::Result<void>`, `CheckPackageLimits` now returns void when package limit is not hit, and returns the error string whenever package limit is hit instead of using out parameter `errString`.\r\n* The PR updates [`checkChainLimits`](https://github.com/bitcoin/bitcoin/blob/5800c558eb5efb4839ed00d6967e43306d68e1c3/src/node/interfaces.cpp#L703) return type to `util::Result<void>`.\r\n\r\n* Now the wallet `CreateTransactionInternal` will have access to the package limit error string whenever its hit.\r\n* Also Updated functional test to reflect the error message from `CTxMempool::CheckPackageLimits` output.\r\n\r\n\r\n",
   "closed_at" : "2023-12-18T15:35:39Z",
   "closed_by" : {
      "avatar_url" : "https://avatars.githubusercontent.com/u/25183001?v=4",
      "events_url" : "https://api.github.com/users/glozow/events{/privacy}",
      "followers_url" : "https://api.github.com/users/glozow/followers",
      "following_url" : "https://api.github.com/users/glozow/following{/other_user}",
      "gists_url" : "https://api.github.com/users/glozow/gists{/gist_id}",
      "gravatar_id" : "",
      "html_url" : "https://github.com/glozow",
      "id" : 25183001,
      "login" : "glozow",
      "node_id" : "MDQ6VXNlcjI1MTgzMDAx",
      "organizations_url" : "https://api.github.com/users/glozow/orgs",
      "received_events_url" : "https://api.github.com/users/glozow/received_events",
      "repos_url" : "https://api.github.com/users/glozow/repos",
      "site_admin" : false,
      "starred_url" : "https://api.github.com/users/glozow/starred{/owner}{/repo}",
      "subscriptions_url" : "https://api.github.com/users/glozow/subscriptions",
      "type" : "User",
      "url" : "https://api.github.com/users/glozow"
   },
   "comments" : 8,
   "comments_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/28863/comments",
   "created_at" : "2023-11-13T16:57:33Z",
   "draft" : false,
   "events_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/28863/events",
   "html_url" : "https://github.com/bitcoin/bitcoin/pull/28863",
   "id" : 1991065461,
   "labels" : [
      {
         "color" : "08a781",
         "default" : false,
         "description" : null,
         "id" : 149424,
         "name" : "Wallet",
         "node_id" : "MDU6TGFiZWwxNDk0MjQ=",
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/labels/Wallet"
      }
   ],
   "labels_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/28863/labels{/name}",
   "locked" : false,
   "milestone" : null,
   "node_id" : "PR_kwDOABII585fUlj4",
   "number" : 28863,
   "performed_via_github_app" : null,
   "pull_request" : {
      "diff_url" : "https://github.com/bitcoin/bitcoin/pull/28863.diff",
      "html_url" : "https://github.com/bitcoin/bitcoin/pull/28863",
      "merged_at" : "2023-12-18T15:35:39Z",
      "patch_url" : "https://github.com/bitcoin/bitcoin/pull/28863.patch",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/pulls/28863"
   },
   "reactions" : {
      "+1" : 0,
      "-1" : 0,
      "confused" : 0,
      "eyes" : 0,
      "heart" : 0,
      "hooray" : 0,
      "laugh" : 0,
      "rocket" : 0,
      "total_count" : 0,
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/28863/reactions"
   },
   "repository_url" : "https://api.github.com/repos/bitcoin/bitcoin",
   "state" : "closed",
   "state_reason" : null,
   "timeline_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/28863/timeline",
   "title" : "wallet, mempool: propagete `checkChainLimits` error message to wallet",
   "updated_at" : "2023-12-18T15:36:01Z",
   "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/28863",
   "user" : {
      "avatar_url" : "https://avatars.githubusercontent.com/u/48946461?v=4",
      "events_url" : "https://api.github.com/users/ismaelsadeeq/events{/privacy}",
      "followers_url" : "https://api.github.com/users/ismaelsadeeq/followers",
      "following_url" : "https://api.github.com/users/ismaelsadeeq/following{/other_user}",
      "gists_url" : "https://api.github.com/users/ismaelsadeeq/gists{/gist_id}",
      "gravatar_id" : "",
      "html_url" : "https://github.com/ismaelsadeeq",
      "id" : 48946461,
      "login" : "ismaelsadeeq",
      "node_id" : "MDQ6VXNlcjQ4OTQ2NDYx",
      "organizations_url" : "https://api.github.com/users/ismaelsadeeq/orgs",
      "received_events_url" : "https://api.github.com/users/ismaelsadeeq/received_events",
      "repos_url" : "https://api.github.com/users/ismaelsadeeq/repos",
      "site_admin" : false,
      "starred_url" : "https://api.github.com/users/ismaelsadeeq/starred{/owner}{/repo}",
      "subscriptions_url" : "https://api.github.com/users/ismaelsadeeq/subscriptions",
      "type" : "User",
      "url" : "https://api.github.com/users/ismaelsadeeq"
   }
}
